torch/__init__.py

```diff
@@ -145,18 +145,23 @@
 
 
 def _preload_cuda_deps():
-    """ Preloads cudnn/cublas deps if they could not be found otherwise """
-    # Should only be called on Linux if default path resolution have failed
-    assert platform.system() == 'Linux', 'Should only be called on Linux'
+    """Preloads cudnn/cublas deps if they could not be found otherwise."""
+    cublas_path = None
+    cudnn_path = None
     for path in sys.path:
         nvidia_path = os.path.join(path, 'nvidia')
         if not os.path.exists(nvidia_path):
             continue
-        cublas_path = os.path.join(nvidia_path, 'cublas', 'lib', 'libcublas.so.11')
-        cudnn_path = os.path.join(nvidia_path, 'cudnn', 'lib', 'libcudnn.so.8')
-        if not os.path.exists(cublas_path) or not os.path.exists(cudnn_path):
-            continue
-        break
+        candidate_cublas_path = os.path.join(nvidia_path, 'cublas', 'lib', 'libcublas.so.11')
+        if os.path.exists(candidate_cublas_path) and not cublas_path:
+            cublas_path = candidate_cublas_path
```

Comment:
```suggestion
        candidate_cublas_path = os.path.join(nvidia_path, 'cublas', 'lib', 'libcublas.so.11')
        if os.path.exists(candidate_cublas_path) and not cublas_path:
            if cublas_path is not None:
               import warnings
               warnings.warn(f"Replacing previosly found cublas {cublas_path} with new candidate {candidate_cublas_path}")
            cublas_path = candidate_cublas_path
```

Reply:
I'm confused by your suggestion. As written the warning will never happen. Did you mean the following?

```python3
if os.path.exists(candidate_cublas_path):
    if cublas path is not None:
        import warnings
        warnings.warn(f"Replacing previosly found cublas {cublas_path} with new candidate {candidate_cublas_path}")
    cublas_path = candidate_cublas_path
```

Reply:
But also I thought the general pattern is to take libraries from earlier in the path if possible, so shouldn't we never do the replacement?

Reply:
@malfet I've pushed just the assert. LMK if you definitely want these warnings.